src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -2,22 +2,24 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.IO;
+using System.Reflection;
 using Microsoft.ML;
 using Microsoft.ML.Data;
 using Microsoft.ML.Data.IO;
 using Microsoft.ML.Runtime;
 
-[assembly: LoadableClass(typeof(BinaryPredictionTransformer<IPredictorProducing<float>>), typeof(BinaryPredictionTransformer), null, typeof(SignatureLoadModel),
+[assembly: LoadableClass(typeof(object), typeof(BinaryPredictionTransformer), null, typeof(SignatureLoadModel),
```

Comment:
This part seems a little concerning. What are the drawbacks of putting `object` here?

Reply:
I believe that it has to be 'object' in there, since the Create method now returns an object. I was working with @codemzs when we decided to put 'object' in there to make it work. I am not sure about the drawbacks.

Reply:
So we agree that it's ok to leave object in the LoadableClassAttribute, right? @codemzs @eerhardt

Reply:
I took 20 minutes to look through the code. The only drawback I could find is when someone internally is trying to create one of these objects through the ComponentCatalog, they will have to ask for it using `typeof(object)` and then cast it down.

However, I didn't find any usages like this in the code when I searched. Since PredictionTransformer is a newer type (newer than the internal TLC code), there is a good chance no one is doing this internally. Also, I believe this only gets called when loading a serialized model, so I think we can make this change.

Reply:
Given @yaeldekel's other comment, we could change this to be `typeof(ISingleFeaturePredictionTransformer<object>)` instead, to give this at least some semblance of strong typing.

Reply:
Thanks very much for taking the time to look at this.

In my latest commit I have updated the `LoadableClassAttribute`s with the `typeof(ISingleFeaturePredictionTransformer<object>)` you requested.